wallet-new/src/Cardano/Wallet/API/V1/Internal/Redemption.hs

```diff
@@ -0,0 +1,57 @@
+module Cardano.Wallet.API.V1.Internal.Redemption where
+
+import           Universum
+
+import           Data.Default (def)
+import           Servant
+
+import           Cardano.Wallet.API.Response
+import           Cardano.Wallet.API.V1.Migration (HasConfigurations, MonadV1,
+                     migrate)
+import           Cardano.Wallet.API.V1.Types
+import           Pos.Core (TxAux)
+import           Pos.Crypto.Configuration (ProtocolMagic)
+import qualified Pos.Wallet.Web.Methods.Redeem as V0
+import qualified Pos.Wallet.Web.ClientTypes as V0
+import qualified Pos.Util.Servant as V0
+
+type API
+    = Summary "TODO: Describe the redemption endpoint"
+    :> "redemptions"
+    :> "ada"
+    :> ReqBody '[ValidJSON] Redemption
+    :> Get '[ValidJSON] (WalletResponse Transaction)
+
+handlers
+    :: HasConfigurations
+    => ProtocolMagic
+    -> (TxAux -> MonadV1 Bool)
+    -> ServerT API MonadV1
+handlers = redeemAda
+
+redeemAda
+    :: HasConfigurations
+    => ProtocolMagic
+    -> (TxAux -> MonadV1 Bool)
+    -> Redemption
+    -> MonadV1 (WalletResponse Transaction)
+redeemAda pm submitTx r = do
+    let ShieldedRedemptionCode seed = redemptionRedemptionCode r
+        passphrase = maybe def unV1 (redemptionPassphrase r)
+    accountId <- migrate (redemptionWalletId r, redemptionAccountIndex r)
+    let caccountId = V0.encodeCType accountId
+    fmap single . migrate =<< case redemptionMnemonic r of
+        Just mnemonic -> do
+            phrase <- migrate mnemonic
```

Comment:
There are quite a few places where I didn't do that and simply applied `CBackupPhrase` to the mnemonic to make them backward compatible, completely forgetting about this `migrate` thingy. I wonder how we could enforce the usage of `migrate` :thinking:

Reply:
I'm not sure it's necessary to try and enforce migrate. Sometimes a one-off explicit change is fine, and this is probably one of those instances. But I just didn't know it'd be that easy :sweat_smile: